compose/runtime/runtime/src/commonTest/kotlin/androidx/compose/runtime/snapshots/SnapshotStateListTests.kt

```diff
@@ -572,124 +573,132 @@ class SnapshotStateListTests {
         }
     }
 
-    @Test(timeout = 10_000)
+    @Test
     @OptIn(ExperimentalCoroutinesApi::class)
     @IgnoreJsTarget // Not relevant in a single threaded environment
-    fun concurrentGlobalModifications_addAll(): Unit = runTest(UnconfinedTestDispatcher()) {
-        repeat(100) {
-            val list = mutableStateListOf<Int>()
-            coroutineScope {
-                repeat(100) { index ->
-                    launch(Dispatchers.Default) {
-                        list.addAll(0, Array(10) { index * 100 + it }.toList())
+    fun concurrentGlobalModifications_addAll() = runTest(UnconfinedTestDispatcher()) {
+        testWithTimeout(10_000) {
```

Comment:
Maybe it makes sense to combine `runTest` with timeout. Is there cases when `testWithTimeout` should be used separately?

Reply:
There is no such case for now.

`runTest` is a function from `kotlinx.coroutines.test`, so we can only wrap it in a new function to extend as you suggest. Do you mean that?

Reply:
If we create `runTest(context, timeout)` function, and change the import, it will look better.

Reply:
Merged. Let me know if any questions/suggestions left